modules/git/command.go

```diff
@@ -47,19 +49,34 @@ func NewCommand(args ...string) *Command {
 	cargs := make([]string, len(GlobalCommandArgs))
 	copy(cargs, GlobalCommandArgs)
 	return &Command{
-		name: GitExecutable,
-		args: append(cargs, args...),
+		name:          GitExecutable,
+		args:          append(cargs, args...),
+		parentContext: DefaultContext,
 	}
 }
 
 // NewCommandNoGlobals creates and returns a new Git Command based on given command and arguments only with the specify args and don't care global command args
 func NewCommandNoGlobals(args ...string) *Command {
 	return &Command{
-		name: GitExecutable,
-		args: args,
+		name:          GitExecutable,
+		args:          args,
+		parentContext: DefaultContext,
 	}
 }
 
+// SetParentContext sets the parent context for this command
+func (c *Command) SetParentContext(ctx context.Context) *Command {
```

Comment:
There's no race. 

* git.Command is not used by multiple goroutines. 

* When the cmd is run a context is created from the parent context at that point. Even if you did change the parentctx at that point it would do nothing to the context of the running process.

Reply:
I dunno if we'll ever use it but it felt appropriate to add it to make it possible for tests to be created in contexts and then cancelled etc.